app/controllers/api/v1x0/stages_controller.rb

```diff
@@ -10,7 +10,22 @@ def show
 
       def index
         req = Request.find(params.require(:request_id))
-        collection(req.stages)
+
+        RBAC::Access.enabled? ? collection(rbac_scope(req.stages)) : collection(req.stages)
+      end
+
+      private
+
+      def rbac_scope(relation)
+        access_obj = RBAC::Access.new('stages', 'read').process
+        return relation if access_obj.admin?
+
+        raise Exceptions::NotAuthorizedError, "Not Authorized to list stages" unless access_obj.owner? || access_obj.accessible?
```

Comment:
Same as actions controller. Check only for `accessible?` and comment below only `owner` is allowed.